src/validation/schema.js

```diff
@@ -389,6 +389,287 @@ const validateNoImportDirective = def =>
+    return errors.concat(
+      ['name', 'language', 'algorithm', 'include'].includes(argument.name.value)
+        ? List([])
+        : List([
+            immutable.fromJS({
+              loc: directive.name.loc,
+              entity: def.name.value,
+              message: `found invalid argument: '${argument.name.value}', @fulltext directives only allow 'name','language', 'algorithm', and 'includes' arguments`,
+            }),
+          ]),
+    )
+  }, List([]))
+}
+
+const validateFullTextDirectiveName = (def, directive) => {
+  let types = directive.arguments.find(argument => argument.name.value == 'name')
+  return types
+    ? validateFullTextDirectiveArgumentName(def, directive, types)
+    : List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'name' must be specified`,
+        }),
+      ])
+}
+
+const validateFullTextDirectiveArgumentName = (def, directive, argument) => {
+  return argument.value.kind != 'StringValue'
+    ? List().push(
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'name' must be a string`,
+        }),
+      )
+    : List([])
+}
+
+const validateFullTextDirectiveLanguage = (def, directive) => {
+  let types = directive.arguments.find(argument => argument.name.value == 'language')
+  return types
+    ? validateFullTextDirectiveArgumentLanguage(def, directive, types)
+    : List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'language' must be specified`,
+        }),
+      ])
+}
+
+const validateFullTextDirectiveArgumentLanguage = (def, directive, argument) => {
+  return argument.value.kind != 'StringValue'
+    ? List().push(
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'language' must be a string`,
+        }),
+      )
+    : List([])
+}
+
+const validateFullTextDirectiveAlgorithm = (def, directive) => {
+  let types = directive.arguments.find(argument => argument.name.value == 'algorithm')
+  return types
+    ? validateFullTextDirectiveArgumentAlgorithm(def, directive, types)
+    : List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'algorithm' must be specified`,
+        }),
+      ])
+}
+
+const validateFullTextDirectiveArgumentAlgorithm = (def, directive, argument) => {
+  if (argument.value.kind != 'EnumValue') {
+    return List().push(
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'algorithm' must be an enum`,
+      }),
+    )
+  } else if (!['ranked', 'proximityRank'].includes(argument.value.value)) {
```

Comment:
Seem that ALLCAPS is the convention for enums: https://spec.graphql.org/draft/#sec-Enum-Value